src/gondi.ts

```diff
@@ -602,11 +604,15 @@ export class Gondi {
           loan: refinancings[0].loan,
           loanId: refinancings[0].loanReferenceId,
         });
-        const refinanceBatch = await contract.refinanceBatch({
-          refinancings,
-          renegotiationId,
-        });
-        results.push(refinanceBatch);
+        try {
+          const refinanceBatch = await contract.refinanceBatch({
+            refinancings,
+            renegotiationId,
+          });
+          results.push({ status: FULFILLED, value: refinanceBatch });
+        } catch (reason) {
+          results.push({ status: REJECTED, reason });
```

Comment:
I'd rather change to this or export the constants
```suggestion
          results.push({ isFulfilled: true, isRejected: false, value: refinanceBatch });
        } catch (reason) {
          results.push({ isFulfilled: false, isRejected: true, reason });
```